File: ios/SsdidWallet/Domain/Verifier/Offline/VerificationBundle.swift
import Foundation

/// A self-contained verification bundle that enables offline credential verification.
/// Contains the issuer's DID Document and optionally a cached status list snapshot.
struct VerificationBundle: Codable {
    let issuerDid: String
    let didDocument: DidDocument
    var statusList: StatusListCredential? = nil
    let fetchedAt: String
    let expiresAt: String
}

/// Storage for verification bundles.
protocol BundleStore: Sendable {
    func saveBundle(_ bundle: VerificationBundle) async throws
    func getBundle(issuerDid: String) async -> VerificationBundle?
    func deleteBundle(issuerDid: String) async throws
    func listBundles() async throws -> [VerificationBundle]
}

File: android/app/src/main/java/my/ssdid/wallet/domain/verifier/offline/VerificationBundle.kt
package my.ssdid.wallet.domain.verifier.offline

import kotlinx.serialization.Serializable
import my.ssdid.sdk.domain.model.DidDocument
import my.ssdid.wallet.domain.revocation.StatusListCredential

/**
 * A self-contained verification bundle that enables offline credential verification.
 * Contains the issuer's DID Document and optionally a cached status list snapshot.
 */
@Serializable
data class VerificationBundle(
    val issuerDid: String,
    val didDocument: DidDocument,
    val statusList: StatusListCredential? = null,
    val fetchedAt: String,
    val expiresAt: String
)
